packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -1286,8 +1290,11 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     if (!props.runtime.supportsCodeGuruProfiling) {
       throw new Error(`CodeGuru profiling is not supported by runtime ${props.runtime.name}`);
     }
-    if (props.environment && (props.environment.AWS_CODEGURU_PROFILER_GROUP_ARN || props.environment.AWS_CODEGURU_PROFILER_ENABLED)) {
-      throw new Error('AWS_CODEGURU_PROFILER_GROUP_ARN and AWS_CODEGURU_PROFILER_ENABLED must not be set when profiling options enabled');
+    if (props.environment && (props.environment.AWS_CODEGURU_PROFILER_GROUP_NAME
+       || props.environment.AWS_CODEGURU_PROFILER_GROUP_ARN
+       || props.environment.AWS_CODEGURU_PROFILER_TARGET_REGION
+       || props.environment.AWS_CODEGURU_PROFILER_ENABLED)) {
```

Comment:
Would adding this have backward incompatible issues?

Reply:
That's fair, though I think it would also be awkward to have some of these be errors and some not be. I've opted to switching this to a warning annotation.